### PR TITLE
[agent-d][issue #229] Unify retry-state ownership across agent receipt paths

### LINKED ISSUE #229: [runtime improvement] Unify retry-state ownership across delivery-backed and receipt-only agent receipt paths
## Objective
Establish one canonical retry-state owner across both delivery-backed and receipt-only agent receipt paths.

## Why
`#223` makes retry advancement atomic and coordinated for agent receipts that have an `event_deliveries` row, with `event_deliveries.retry_count` as the canonical retry owner in that seam. But receipt-only agent paths still keep retry count in the receipt row because there is no delivery row to own that state.

That leaves one core semantic split still in the platform:
- delivery-backed agent receipt paths use `event_deliveries` as the retry owner
- receipt-only agent paths use `event_receipts` side effects as the retry owner

The atomicity fix is correct and mergeable as-is. This follow-up is about removing the remaining split semantic owner if the platform wants one universal retry model.

## In scope
- decide the canonical retry-state owner for receipt-only agent processing
- either eliminate receipt-only agent processing in the touched seam or normalize it onto persisted delivery rows everywhere
- update read/write paths in the touched seam so retry ownership is singular
- add focused tests proving retry semantics stay aligned across both paths after the boundary is unified

## Out of scope
- the atomic write fix already delivered by `#223`
- unrelated delivery lifecycle redesign
- spec changes unless a real ambiguity appears

## Constraints
- preserve fail-fast and fail-closed behavior
- keep one canonical retry owner for the touched agent receipt seam
- do not reintroduce split retry ownership through compatibility branches or fallback paths
- keep the change scoped to retry ownership, not broader runtime/store redesign

## Acceptance criteria
- one canonical retry-state owner exists across delivery-backed and receipt-only agent receipt paths in the touched seam
- read and write paths no longer depend on separate retry owners for those two cases
- focused tests prove equivalent retry behavior across both path types

## Spec References

- `docs/specs/swarm-platform/platform/contracts/platform-spec.yaml: event_schema.routing_derivation.delivery_rules`
- `docs/specs/swarm-platform/platform/contracts/platform-spec.yaml: error_model.handler_execution_failure.retry_policy`
- `docs/specs/swarm-platform/platform/contracts/platform-spec.yaml: error_model.dead_letter_schema`
- governing rule: delivery, retry, pending, and dead-letter behavior must come from one canonical delivery owner; runtime/store/operator surfaces must not invent alternate recipient or retry semantics.

### PR DESCRIPTION
**Pull request by yazzaoui**
Closes #229

## Spec references
- `docs/specs/swarm-platform/platform/contracts/platform-spec.yaml: event_schema.routing_derivation.delivery_rules`
- `docs/specs/swarm-platform/platform/contracts/platform-spec.yaml: error_model.handler_execution_failure.retry_policy`
- `docs/specs/swarm-platform/platform/contracts/platform-spec.yaml: error_model.dead_letter_schema`
- governing rule: delivery, retry, pending, and dead-letter behavior in the touched seam must come from one canonical delivery owner rather than separate receipt-side and delivery-side retry interpretations.

## Human Summary
This PR removes the split retry owner for the first covered agent receipt slice. Agent receipt failures now normalize onto a delivery row before retry progression is evaluated, so equivalent retryable and exhausted cases reach the same canonical state whether the event started delivery-backed or not.

## What Changed
- normalized agent receipt retry progression onto `event_deliveries` inside `UpsertEventReceipt(...)`
- removed the receipt-only pending retry/backoff branch from canonical pending selection
- added focused coverage proving delivery-backed and previously receipt-only agent paths now land on the same retryable and exhausted outcomes

## Why This Is Needed
The touched seam was still letting retry-state meaning split between `event_deliveries.retry_count` and receipt-side `side_effects.retry_count`. That made retryable vs exhausted behavior depend on which path handled the failure. This PR gives the first covered slice one clearer canonical retry owner and aligns the store/runtime meaning around it.

## Scope Boundaries
- keeps the change narrow to the touched agent receipt retry seam
- does not redesign receipt-only non-agent paths
- does not broaden into delivery lifecycle observability or other recovery semantics beyond the retry-owner slice needed here

## Tests Run
- `go test ./internal/store -run 'Test(UpsertEventReceipt_(PreservesRetryableVsTerminalDeliveryStatus_V2|AlignsRetryOwnershipAcrossDeliveryBackedAndReceiptOnly_V2)|ListPendingEventsForAgent_RetryBackoff_V2|ListPendingSubscribedEvents_RetryBackoff_V2)' -count=1`
- `go test ./internal/store ./internal/runtime/manager -count=1`
- `go test ./... -count=1`

## Residual Risk
Receipt-only paths that are not agent delivery-backed still keep their own semantics because they do not participate in `event_deliveries`. This PR makes that boundary sharper for the first covered agent slice, but it does not redesign every receipt surface.

## Follow-Up
If additional agent paths still bypass delivery ownership entirely, they should be normalized onto the same canonical delivery-backed retry owner rather than reintroducing receipt-side retry progression.

### CODE REVIEW

**Comment by yazzaoui** [comment]
## Self-Audit

Spec re-read:
- `docs/specs/swarm-platform/platform/contracts/platform-spec.yaml: event_schema.routing_derivation.delivery_rules`
- `docs/specs/swarm-platform/platform/contracts/platform-spec.yaml: error_model.handler_execution_failure.retry_policy`
- `docs/specs/swarm-platform/platform/contracts/platform-spec.yaml: error_model.dead_letter_schema`
- I re-read those sections before implementation and again before posting this audit. The final code matches them for the first covered slice: delivery-backed agent retry progression is now authoritative for equivalent retryable and exhausted cases, and dead-letter transition meaning is no longer path-dependent in the touched seam.

Canonical owner:
- Single canonical owner: `event_deliveries.retry_count` plus delivery status/reason in the store receipt path for the touched agent retry slice.
- Adjacent surfaces checked against it:
  - `internal/store/event_receipt_store.go`
  - `internal/store/pending_delivery_selection.go`
  - `internal/store/events.go`
  - `internal/runtime/manager/receipts.go`
  - `internal/store/manager_retry_policy_test.go`

Counterexamples checked:
- Broader failure class: yes, the reported bug was symptom-shaped. The real local failure class was split retry ownership between delivery-backed and receipt-only agent receipt paths, plus pending-selection semantics that still interpreted receipt-side retry state independently.
- Sibling contexts checked under the same rule:
  - delivery-backed `UpsertEventReceipt(...)`
  - receipt-only agent `UpsertEventReceipt(...)`
  - canonical pending selection for agent deliveries
  - runtime manager receipt/read usage in `internal/runtime/manager/receipts.go`
- Main invalid / rollback / mixed-case / fail-closed cases considered:
  - same retryable error applied first to a delivery-backed event and to a receipt-only event
  - same exhausted error applied on the second failure to both paths
  - pending-query drift if receipt-only retry/backoff remained active after normalizing onto deliveries
  - fail-closed behavior if a delivery retry owner cannot be created for an agent receipt path
- Focused tests added or updated:
  - `TestUpsertEventReceipt_AlignsRetryOwnershipAcrossDeliveryBackedAndReceiptOnly_V2`
  - existing touched regressions re-run:
    - `TestUpsertEventReceipt_PreservesRetryableVsTerminalDeliveryStatus_V2`
    - `TestListPendingEventsForAgent_RetryBackoff_V2`
    - `TestListPendingSubscribedEvents_RetryBackoff_V2`
- Generic failing proof created: cross-path equivalence test proving delivery-backed and previously receipt-only agent failures now land on the same retryable and exhausted outcomes.

Scope check:
- Adjacent off-spec behavior noticed but not changed:
  - receipt-only paths outside this touched agent seam still keep their own semantics where no `event_deliveries` row participates.
- Uncaptured spec gap / ambiguity / contradiction / off-spec behavior outside scope:
  - none found in the touched seam, so no escalation was needed.

Residual interaction risk:
- Most likely hidden interaction worth reviewer attention: any future agent receipt entrypoint that bypasses `UpsertEventReceipt(...)` and writes receipt-side retry state directly could reintroduce split ownership unless it is normalized onto the same delivery-backed store path.

**Comment by yazzaoui** [comment]
## Post-Implementation Proof Audit

Broader failure class
- Split retry-state ownership across agent receipt handling, where equivalent delivery failures could derive retryable vs exhausted meaning from different persisted owners depending on whether the path already had an `event_deliveries` row.

Issue shape
- The issue described a real broader class, not just one isolated symptom. The implementation patch stayed narrow, but the semantic assessment covered the adjacent pending-selection and receipt-consumer surfaces under the same retry-policy rule.

Sibling contexts checked
- delivery-backed `UpsertEventReceipt(...)`
- previously receipt-only agent `UpsertEventReceipt(...)`
- canonical pending selection for agent deliveries
- adjacent runtime manager receipt/read consumers in `internal/runtime/manager/receipts.go`

Generic failing proof used or created
- Created `TestUpsertEventReceipt_AlignsRetryOwnershipAcrossDeliveryBackedAndReceiptOnly_V2` as the generic cross-path equivalence reproducer for the failure class.

Failure-class coverage table

| Manifestation | Final status | Exact proof used | Mandatory end-to-end / supported-surface proof | Notes |
| --- | --- | --- | --- | --- |
| Delivery-backed agent receipt path uses `event_deliveries.retry_count` | reproduced and fixed | `TestUpsertEventReceipt_AlignsRetryOwnershipAcrossDeliveryBackedAndReceiptOnly_V2` and `TestUpsertEventReceipt_PreservesRetryableVsTerminalDeliveryStatus_V2` | `go test ./... -count=1` | Delivery-backed path remains canonical and unchanged in meaning. |
| Receipt-only agent receipt path uses receipt-side `side_effects.retry_count` | reproduced and fixed | `TestUpsertEventReceipt_AlignsRetryOwnershipAcrossDeliveryBackedAndReceiptOnly_V2` | `go test ./... -count=1` | This path now normalizes onto a delivery row before retry progression is evaluated. |
| Canonical pending selection still interprets receipt-side retry/backoff independently when no delivery row exists | execution-proven through the same corrected path | `TestListPendingEventsForAgent_RetryBackoff_V2` and `TestListPendingSubscribedEvents_RetryBackoff_V2` after the store normalization change | `go test ./... -count=1` | Agent retry/backoff now follows delivery ownership in the touched seam; no receipt-only retry branch remains for agent pending selection. |
| Runtime manager receipt/read consumers could observe divergent retry meaning if store owners remain split | execution-proven through the same corrected path | `go test ./internal/store ./internal/runtime/manager -count=1` plus the focused store equivalence reproducer above | `go test ./... -count=1` | No consumer-specific rewrite was needed once the store owner was unified for the covered slice. |

Unproven manifestations
- none among the currently known manifestations in scope

Residual interaction risk
- Most likely hidden interaction worth reviewer attention: any future agent receipt entrypoint that bypasses `UpsertEventReceipt(...)` and writes receipt-side retry state directly could reintroduce split ownership unless it is normalized onto the same delivery-backed store path.

File: internal/store/event_receipt_store.go [hunk 1]
```diff
@@ -294,27 +285,23 @@ func (s *PostgresStore) applyDeliveryBackedTerminalTransitionTx(
...
-	case err != nil:
-		return 0, fmt.Errorf("load event receipt side effects: %w", err)
+func (s *PostgresStore) ensureAgentDeliveryRetryOwnerTx(
+	ctx context.Context,
+	caps StoreSchemaCapabilities,
+	tx *sql.Tx,
+	eventID, agentID string,
+) (lockedAgentDelivery, error) {
+	if err := s.insertEventDeliveriesSpec(ctx, caps, tx, eventID, []string{agentID}); err != nil {
+		return lockedAgentDelivery{}, err
```
[the remark this answers is not in the document]
Comment (yazzaoui) [reply internal/store/event_receipt_store.go:314]: Still valid on current head `38702210448b0cbc155fcafe9f057cdb22628438`. `ensureAgentDeliveryRetryOwnerTx(...)` still inserts a fresh `event_deliveries` row and never seeds it from any existing receipt-side retry history. `buildAgentReceiptWriteState(...)` then derives the next state from `delivery.retry_count`, so a legacy agent receipt row with `side_effects.retry_count > 0` and no delivery row still resets to `0 -> 1` instead of exhausting on the next failure.

File: internal/store/pending_delivery_selection.go [hunk 2]
```diff
@@ -30,14 +30,7 @@ func CanonicalPendingAgentDeliveryPredicateSQL(deliveryAlias, receiptAlias strin
...
-			AND (
-				%s.event_id IS NULL
-				OR (
-					COALESCE(%s.side_effects->>'manager_status', '') = 'error'
-					AND COALESCE((%s.side_effects->>'retry_count')::int, 0) < 2
-					AND %s.processed_at <= now() - interval '1 minute'
-				)
-			)
+			AND %s.event_id IS NULL
 		)
```
[the remark this answers is not in the document]
Comment (yazzaoui) [reply internal/store/pending_delivery_selection.go:34]: Still valid on current head `38702210448b0cbc155fcafe9f057cdb22628438`. The canonical predicate now explicitly excludes `delivery_id IS NULL` whenever a receipt row exists, and `ListPendingEventsForAgent(...)` also reads only from `event_deliveries`, so legacy agent events with a receipt row, no delivery row, and elapsed backoff still disappear from both pending queries instead of being selected for retry or normalized first.

**Comment by yazzaoui** [comment]
Reviewed against issue `#229`, the issue thread, the cited spec sections (`delivery_rules`, `retry_policy`, `dead_letter_schema`), the issue-level pre-implementation coverage audit, and the PR post-implementation proof audit on current head `38702210448b0cbc155fcafe9f057cdb22628438`.

I am not approving this PR on the current head. Two unresolved P1 inline blockers are still valid:

1. [`event_receipt_store.go`](internal/store/event_receipt_store.go) still resets legacy receipt-only retry history when normalizing onto a new delivery row. `ensureAgentDeliveryRetryOwnerTx(...)` inserts a fresh `event_deliveries` row, `buildAgentReceiptWriteState(...)` derives from `delivery.retry_count`, and no migration step seeds that count from an existing receipt-side `side_effects.retry_count`. For agent receipts written by the old path, the next failure still grants extra retries instead of exhausting as expected.

2. [`pending_delivery_selection.go`](internal/store/pending_delivery_selection.go) still drops retryable legacy receipt-only agent rows from both pending queries. The new predicate excludes `delivery_id IS NULL` whenever a receipt row exists, and `ListPendingEventsForAgent(...)` already reads only from `event_deliveries`, so legacy rows with backoff elapsed and no delivery row are not selectable for retry unless something else rewrites them first.

The process artifacts are present and materially better than the older failure-class misses: the pre-audit is specific, the PR proof audit has a real coverage table, and the issue framing is appropriate for a first slice. But the touched implementation still fails legacy-state continuity inside the exact seam it claims to normalize. That means the current head is not yet compliant with the one-canonical-owner goal for the covered agent receipt slice.

I would re-check this PR once legacy receipt-only state is either migrated into the new delivery owner correctly or explicitly handled fail-closed without granting extra retries or orphaning pending work.

**Comment by yazzaoui** [comment]
Review checklist for current head `38702210448b0cbc155fcafe9f057cdb22628438`:

- Guideline/process check: issue read, issue thread read, spec refs checked, pre-audit present, PR proof audit present
- Failure-class framing: good; `#229` correctly targets split retry-owner semantics rather than one isolated symptom
- Failure-class coverage: not complete on current head because legacy receipt-only rows without a delivery record still regress in retry progression and pending selection
- Residual risk: old receipt-only agent rows can gain extra retries after normalization and can disappear from pending replay selection entirely
- Risk level: high
- Merge recommendation: do not merge until the two unresolved P1 inline blockers are fixed and re-tested
- Follow-up tracking: no new issue needed; both blockers are already captured by the existing unresolved inline review threads
- Spec compliance assessment: not yet sufficient for the touched seam on the current head

File: internal/store/event_receipt_store.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (yazzaoui) [reply internal/store/event_receipt_store.go:314]: Fixed on `b1651c2`.

`ensureAgentDeliveryRetryOwnerTx(...)` now locks any existing agent receipt row, carries its decoded retry-state into the canonical delivery row, and backfills the missing `event_deliveries` state before the next retry decision is derived. That preserves legacy receipt-only retry history instead of resetting it to `0 -> 1`.

Focused proof added:
- `TestUpsertEventReceipt_PreservesLegacyReceiptOnlyRetryHistory_V2`
- `go test ./internal/store -run 'Test(UpsertEventReceipt_(PreservesRetryableVsTerminalDeliveryStatus_V2|AlignsRetryOwnershipAcrossDeliveryBackedAndReceiptOnly_V2|PreservesLegacyReceiptOnlyRetryHistory_V2)|ListPendingEventsForAgent_RetryBackoff_V2|ListPendingSubscribedEvents_RetryBackoff_V2|PendingAgentEvents_NormalizesLegacyReceiptOnlyRetryOwner_V2)' -count=1`
- `go test ./internal/store ./internal/runtime/manager -count=1`
- `go test ./... -count=1`

File: internal/store/pending_delivery_selection.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (yazzaoui) [reply internal/store/pending_delivery_selection.go:34]: Fixed on `b1651c2`.

I kept the canonical pending predicate strict and moved the legacy continuity fix to the store boundary instead: `ListPendingEventsForAgent(...)` and `ListPendingSubscribedEvents(...)` now normalize legacy agent receipt rows onto canonical `event_deliveries` before evaluating pending eligibility. That keeps retry-state ownership singular while preserving retryable legacy rows long enough to be retried through the canonical owner.

Focused proof added:
- `TestPendingAgentEvents_NormalizesLegacyReceiptOnlyRetryOwner_V2`
- `go test ./internal/store -run 'Test(UpsertEventReceipt_(PreservesRetryableVsTerminalDeliveryStatus_V2|AlignsRetryOwnershipAcrossDeliveryBackedAndReceiptOnly_V2|PreservesLegacyReceiptOnlyRetryHistory_V2)|ListPendingEventsForAgent_RetryBackoff_V2|ListPendingSubscribedEvents_RetryBackoff_V2|PendingAgentEvents_NormalizesLegacyReceiptOnlyRetryOwner_V2)' -count=1`
- `go test ./internal/store ./internal/runtime/manager -count=1`
- `go test ./... -count=1`

**Comment by yazzaoui** [comment]
## Post-Implementation Proof Audit Addendum

The two legacy-state manifestations raised in review are now covered on head `b1651c2b5d586cf340c02da1fd82fafed45e425f`.

Additional manifestation coverage:

| Manifestation | Final status | Exact proof used | Mandatory end-to-end / supported-surface proof | Notes |
| --- | --- | --- | --- | --- |
| Legacy receipt-only agent row with `side_effects.retry_count > 0` resets to `0 -> 1` when normalized onto a new delivery row | reproduced and fixed | `TestUpsertEventReceipt_PreservesLegacyReceiptOnlyRetryHistory_V2` | `go test ./... -count=1` | Missing delivery rows are now backfilled from locked receipt-side retry history before the next retry decision is derived. |
| Legacy retryable agent receipt row with no delivery record disappears from pending selection instead of being retried or normalized first | reproduced and fixed | `TestPendingAgentEvents_NormalizesLegacyReceiptOnlyRetryOwner_V2` | `go test ./... -count=1` | Pending read paths now normalize legacy agent receipt rows onto canonical `event_deliveries` before applying the strict canonical pending predicate. |

Updated focused test set:
- `go test ./internal/store -run 'Test(UpsertEventReceipt_(PreservesRetryableVsTerminalDeliveryStatus_V2|AlignsRetryOwnershipAcrossDeliveryBackedAndReceiptOnly_V2|PreservesLegacyReceiptOnlyRetryHistory_V2)|ListPendingEventsForAgent_RetryBackoff_V2|ListPendingSubscribedEvents_RetryBackoff_V2|PendingAgentEvents_NormalizesLegacyReceiptOnlyRetryOwner_V2)' -count=1`
- `go test ./internal/store ./internal/runtime/manager -count=1`
- `go test ./... -count=1`

File: internal/store/event_receipt_store.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (yazzaoui) [reply internal/store/event_receipt_store.go:314]: Obsolete on current head `b1651c2b5d586cf340c02da1fd82fafed45e425f`.

`ensureAgentDeliveryRetryOwnerTx(...)` now locks any existing agent receipt row, decodes its receipt-side retry history, and backfills the new `event_deliveries` row before the next retry decision is derived. That preserves legacy receipt-only retry count instead of resetting it to `0 -> 1`.

Focused proof rechecked locally:
- `TestUpsertEventReceipt_PreservesLegacyReceiptOnlyRetryHistory_V2`
- `go test ./internal/store -run 'Test(UpsertEventReceipt_(PreservesRetryableVsTerminalDeliveryStatus_V2|AlignsRetryOwnershipAcrossDeliveryBackedAndReceiptOnly_V2|PreservesLegacyReceiptOnlyRetryHistory_V2)|ListPendingEventsForAgent_RetryBackoff_V2|ListPendingSubscribedEvents_RetryBackoff_V2|PendingAgentEvents_NormalizesLegacyReceiptOnlyRetryOwner_V2)' -count=1`
- `go test ./internal/store ./internal/runtime/manager -count=1`

File: internal/store/pending_delivery_selection.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (yazzaoui) [reply internal/store/pending_delivery_selection.go:34]: Obsolete on current head `b1651c2b5d586cf340c02da1fd82fafed45e425f`.

The strict canonical pending predicate is still in place, but `ListPendingEventsForAgent(...)` and `ListPendingSubscribedEvents(...)` now normalize legacy agent receipt rows onto canonical `event_deliveries` before pending eligibility is evaluated. That keeps retry-state ownership singular while preserving retryable legacy rows long enough to be selected and retried through the delivery owner.

Focused proof rechecked locally:
- `TestPendingAgentEvents_NormalizesLegacyReceiptOnlyRetryOwner_V2`
- `go test ./internal/store -run 'Test(UpsertEventReceipt_(PreservesRetryableVsTerminalDeliveryStatus_V2|AlignsRetryOwnershipAcrossDeliveryBackedAndReceiptOnly_V2|PreservesLegacyReceiptOnlyRetryHistory_V2)|ListPendingEventsForAgent_RetryBackoff_V2|ListPendingSubscribedEvents_RetryBackoff_V2|PendingAgentEvents_NormalizesLegacyReceiptOnlyRetryOwner_V2)' -count=1`
- `go test ./internal/store ./internal/runtime/manager -count=1`

**Comment by yazzaoui** [comment]
Reviewed against issue `#229`, the issue thread, the cited spec sections (`delivery_rules`, `retry_policy`, `dead_letter_schema`), the existing pre-audit / post-audit artifacts, and current head `b1651c2b5d586cf340c02da1fd82fafed45e425f`.

I do not have substantive findings on the current head.

The two earlier legacy-state blockers are now obsolete:

1. Legacy receipt-only retry history is preserved when a missing delivery owner is normalized. `ensureAgentDeliveryRetryOwnerTx(...)` now locks any existing agent receipt row, decodes its receipt-side retry state, and backfills the canonical `event_deliveries` row before the next retry decision is derived.
2. Legacy retryable agent rows without a delivery record remain selectable long enough to be retried through the canonical owner. `ListPendingEventsForAgent(...)` and `ListPendingSubscribedEvents(...)` now normalize legacy agent receipt rows onto `event_deliveries` before applying the strict canonical pending predicate.

Proof rechecked locally:
- `go test ./internal/store -run 'Test(UpsertEventReceipt_(PreservesRetryableVsTerminalDeliveryStatus_V2|AlignsRetryOwnershipAcrossDeliveryBackedAndReceiptOnly_V2|PreservesLegacyReceiptOnlyRetryHistory_V2)|ListPendingEventsForAgent_RetryBackoff_V2|ListPendingSubscribedEvents_RetryBackoff_V2|PendingAgentEvents_NormalizesLegacyReceiptOnlyRetryOwner_V2)' -count=1`
- `go test ./internal/store ./internal/runtime/manager -count=1`
- `go test ./... -count=1`

Old-process vs new-process note:
- This PR opened under the older process, so I am not treating absence of the newer canonical-owner/systematic-consumption audit shape as a standalone blocker.
- On substance, I do **not** think this now requires re-engineering under the newer process. The touched agent receipt seam has one canonical retry owner on current head, and the remaining legacy read-path normalization is a bounded continuity bridge into that owner rather than a sign of unresolved split ownership in the covered slice.
- If broader receipt-only non-agent paths or new agent entrypoints still bypass this store path, that should be handled as separate follow-on work rather than blocking this PR.

**Comment by yazzaoui** [comment]
Review checklist for current head `b1651c2b5d586cf340c02da1fd82fafed45e425f`:

- Guideline/process check: issue read, issue thread read, spec refs checked, existing pre-audit present, existing PR proof audit present
- Failure-class framing: good; `#229` correctly targets split retry-owner semantics across delivery-backed and receipt-only agent receipt paths
- Failure-class coverage: sufficient for the touched agent slice on current head, including the two legacy continuity manifestations raised during review
- Tests rechecked locally:
  - `go test ./internal/store -run 'Test(UpsertEventReceipt_(PreservesRetryableVsTerminalDeliveryStatus_V2|AlignsRetryOwnershipAcrossDeliveryBackedAndReceiptOnly_V2|PreservesLegacyReceiptOnlyRetryHistory_V2)|ListPendingEventsForAgent_RetryBackoff_V2|ListPendingSubscribedEvents_RetryBackoff_V2|PendingAgentEvents_NormalizesLegacyReceiptOnlyRetryOwner_V2)' -count=1`
  - `go test ./internal/store ./internal/runtime/manager -count=1`
  - `go test ./... -count=1`
- Residual risk: broader receipt-only paths outside the covered agent seam still remain out of scope
- Risk level: medium-low
- Merge recommendation: acceptable to merge
- New-process escalation check: no forced re-engineering needed for this PR; the current solution closes the touched semantic gap tightly enough for the covered slice
- Follow-up tracking: no new issue required from this review pass
- Spec compliance assessment: sufficient for the touched retry-owner seam on the current head
